timemachine/cpp/src/bound_potential.cu

```diff
@@ -58,15 +55,28 @@ void BoundPotential::execute_host(
     }
 };
 
+void BoundPotential::set_params(const std::vector<double> &params) {
+    if (params.size() > 0) {
+        if (params.size() * sizeof(*d_p.data) > d_p.size) {
+            throw std::runtime_error(
+                "parameter size is greater than device buffer size: " + std::to_string(params.size()) + " > " +
+                std::to_string(d_p.size / sizeof(*d_p.data)));
+        }
+        // NOTE: can't use d_p.copy_from here because new parameters size may be less than buffer size
+        gpuErrchk(cudaMemcpyAsync(d_p.data, params.data(), params.size() * sizeof(*d_p.data), cudaMemcpyHostToDevice));
```

Comment:
Also - why is this calling the Async version?

Reply:
Fixed as a side-effect of https://github.com/proteneer/timemachine/pull/1119/commits/4f75acadfd7324d4416d6efac0ad1d59c3b8a8de